io/plugins/eu.esdihumboldt.hale.io.xls/src/eu/esdihumboldt/hale/io/xls/reader/XLSInstanceReader.java

```diff
@@ -165,24 +169,18 @@ private void loadSheet(SheetInfo sheet, IOReporter reporter) throws Exception {
 		Collection<List<String>> rows = analyser.getRows();
 
 		int line = 0;
-		int skipNlines = sheet.getSettings().getSkipLines() != null
-				? sheet.getSettings().getSkipLines()
-				: 0;
 
 		if (skipNlines <= 0) {
-			// do not skip any lines
-
 			// the header row is added as instance
 			addInstanceForRow(analyser.getHeader(), type, propAr, line, reporter);
-
 			line++;
-
 			// iterate over all rows to create the instances
 			Iterator<List<String>> allRows = rows.iterator();
 			while (allRows.hasNext()) {
 				List<String> row = allRows.next();
-				addInstanceForRow(row, type, propAr, line, reporter);
-
+				if (row != null && !row.stream().allMatch(s -> s == null || s.isEmpty())) {
+					addInstanceForRow(row, type, propAr, line, reporter);
+				}
 				line++;
 			}
 		}
```

Comment:
Related to my comment [here](https://github.com/halestudio/hale/pull/1087/files#r1425178823), the first point would imply removing this whole section.